Fix(upload): Make sure _fetchRevisionManifest works if the manifest is empty

fsmanuel opened the pull request:
If there was a previous error writing the revisions manifest file the deployment would be stuck because of an unhandled JSON.parse error. I used the promises of the client instead of creating new ones. Now we get error messages even for the success callbacks.

Please don't merge yet. Because the error is "kind of random" I'll test more with bad latency and let you know if it works.

fsmanuel commented:
@arenoir It worked the last 2 weeks. Should I squash the commits?

arenoir commented:
@fsmanuel,  yes squash it.  I was just going through the changes.  It looks good one thing I noticed was the removal of promise reject handling and error logging.  What is your reasoning for that?  Should we chain a catch function?

fsmanuel committed:
Fix(upload): Make sure _fetchRevisionManifest works if the manifest is empty

If there was a previous error writing the revisions manifest file the deployment would be stuck because of an unhandled JSON.parse error. I used the promises of the client instead of creating new ones. Now we get error messages even for the success callbacks.

fsmanuel commented:
@arenoir Can you comment on the lines you think I forgot error logging? I moved the errors but tried not to remove them.

arenoir commented:
Never mind i miss read the diff.

fsmanuel commented:
@arenoir can we release a new version?

arenoir commented:
Will do this week... have been on vacation.

arenoir commented:
@fsmanuel I released a new version
